ipfs/provider.go

```diff
@@ -4,7 +4,44 @@ import (
 	"io"
 
 	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/ipfs/interface-go-ipfs-core/options"
 )
 
 // APIProvider allows customizable IPFS core APIs.
-type APIProvider func() (coreiface.APIDagService, io.Closer, error)
+type APIProvider func(online bool) (coreiface.APIDagService, io.Closer, error)
+
+type onlineOfflineProvider struct {
+	online, offline coreiface.APIDagService
+	closer          io.Closer
+}
+
+func NewOnlineOfflineProvider(online coreiface.CoreAPI, closer io.Closer) (APIProvider, error) {
+
+	offline, err := online.WithOptions(OfflineOption)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just for my understanding: this is the exact same ipfs node but the offline one has a few changes that strip away the networking. Does `online.WithOptions` create a copy of the original object? Can they co-exist without any conflict between one or the other?

Reply:
This was answered in a sync call. Yes, they can.